### PR TITLE
Add server controlled task queue max dispatch rate

### PR DESCRIPTION
**Pull request by wxing1292**
**What changed?**
* Add admin dynamic config controlled task queue max rate limit

**Why?**
N/A

**How did you test it?**
N/A

**Potential risks**
N/A

### CODE REVIEW

File: common/service/dynamicconfig/config.go [hunk 1]
```diff
@@ -101,6 +101,9 @@ type FloatPropertyFn func(opts ...FilterOption) float64
 // FloatPropertyFnWithShardIDFilter is a wrapper to get float property from dynamic config with shardID as filter
 type FloatPropertyFnWithShardIDFilter func(shardID int32) float64
 
+// FloatPropertyFnWithTaskQueueInfoFilters is a wrapper to get int property from dynamic config with three filters: namespace, taskQueue, taskType
```
Comment (mastermanu) [inline common/service/dynamicconfig/config.go:104]: [Nit] get float property

File: service/matching/config.go [hunk 2]
```diff
@@ -113,6 +115,7 @@ func NewConfig(dc *dynamicconfig.Collection) *Config {
 		ThrottledLogRPS:                 dc.GetIntProperty(dynamicconfig.MatchingThrottledLogRPS, 20),
 		NumTaskqueueWritePartitions:     dc.GetIntPropertyFilteredByTaskQueueInfo(dynamicconfig.MatchingNumTaskqueueWritePartitions, dynamicconfig.DefaultNumTaskQueuePartitions),
 		NumTaskqueueReadPartitions:      dc.GetIntPropertyFilteredByTaskQueueInfo(dynamicconfig.MatchingNumTaskqueueReadPartitions, dynamicconfig.DefaultNumTaskQueuePartitions),
+		TaskqueuePartitionDispatchRate:  dc.GetFloatPropertyFilteredByTaskQueueInfo(dynamicconfig.MatchingTaskqueuePartitionDispatchRate, 2000),
```
Comment (mastermanu) [inline service/matching/config.go:118]: fyi, default out-of-box is 4, which makes it 500 per partition. just want to make sure this won't cause any unexpected behavior if someone upgrades

File: common/service/dynamicconfig/config.go [hunk 3]
```diff
@@ -260,6 +263,37 @@ func (c *Collection) GetFloat64PropertyFilteredByShardID(key Key, defaultValue f
...
+			val, err = c.client.GetFloatValue(
+				key,
+				filterMap,
+				defaultValue,
+			)
+			if err != nil {
+				c.logError(key, err)
+			}
+
+			if val != defaultValue {
```
Comment (mastermanu) [inline common/service/dynamicconfig/config.go:287]: fyi, this is float64, right? equals/not equals could lead to some weird behavior depending on precision

File: common/service/dynamicconfig/config.go [hunk 4]
```diff
@@ -260,6 +263,37 @@ func (c *Collection) GetFloat64PropertyFilteredByShardID(key Key, defaultValue f
...
+				break
+			}
+		}
+
+		c.logValue(key, val, defaultValue, intCompareEquals)
```
Comment (mastermanu) [inline common/service/dynamicconfig/config.go:292]: float64Equals?

**Review by mastermanu — APPROVED** [review]

File: service/matching/config.go (hunk 2 above)
Comment (wxing1292) [reply service/matching/config.go:118]: 500 per partition should be fairly large

File: common/service/dynamicconfig/config.go (hunk 3 above)
Comment (wxing1292) [reply common/service/dynamicconfig/config.go:287]: guess for dynamic config, this is fine. there should be no N/A case, or precision related case

**Pull request merged**